Assets/Scripts/Map/Generators/IMapConfig.cs

```diff
@@ -0,0 +1,11 @@
+
+using UnityEngine;
+
+namespace Maes.Map.Generators
+{
+    public interface IMapConfig
+    {
+        public abstract SimulationMap<Tile> GenerateMap(GameObject gameObject, float wallHeight = 2.0f);
+        public int MapSize { get; }
```

Comment:
OPTIONAL: Is it better to have both the height and width as property, and then in our case we could just us one of them. By doing that we can also ensure in the future that is also would work if the maps are rectangular shapes, instead of square shape 😄